File: VB/E5020/Models/ReportModel.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports DevExpress.XtraReports.UI
Imports E5020.Reports

Namespace E5020.Models
	Public Class ReportModel
		Private privateReportName As String
		Public Property ReportName() As String
			Get
				Return privateReportName
			End Get
			Set(ByVal value As String)
				privateReportName = value
			End Set
		End Property

		Public Function CreateReport() As XtraReport
			Select Case ReportName
				Case "Categories"
					Return New CategoriesReport()
				Case "Products"
					Return New ProductsReport()
				Case "Orders"
					Return New OrdersReport()
			End Select
			Return Nothing
		End Function
	End Class
End Namespace

File: CS/E5020/Models/ReportModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using DevExpress.XtraReports.UI;
using E5020.Reports;

namespace E5020.Models {
    public class ReportModel {
        public string ReportName {
            get;
            set;
        }

        public XtraReport CreateReport() {
            switch (ReportName) {
                case "Categories":
                    return new CategoriesReport();
                case "Products":
                    return new ProductsReport();
                case "Orders":
                    return new OrdersReport();
            }
            return null;
        }
    }
}